### DISCUSSION TITLE
Auto-redirect to link url after power on

### DISCUSSION THREAD

**Discussion by npelov** [seriousm4x/UpSnap#847] [Feature requests]
Hey, just found that useful program. I just love to not have to spin up docker or install ton of dependencies. 

I just got a bunch of small disks for cheap and I got an idea to make it an offline nas with raspi or something. I already have bigger online nas. **It would be nice to have a checkbox "redirect to link on power on" which will redirect on port status change. That should be only if it goes from offline to online and not if it is online while loading the page - that would break the page :).**

Or maybe only make the power button clickable so you can click on ports while the system is offline which will wake it up and redirect as soon as it's available

here is my systemd service - you can add it to the readme if you want

Install:
```
sudo useradd upsnap
sudo mkdir -p /usr/local/upsnap/var/upsnap /usr/local/upsnap/bin
sudo chown -R upsnap /usr/local/upsnap/var/upsnap
sudo cp -a upsnap /usr/local/upsnap/bin/
sudo setcap cap_net_raw=+ep /usr/local/upsnap/bin/upsnap

```
Systemd Service:
```
[Unit]
Description=UpSnap Service
After=network.target

[Service]
Type=simple
User=upsnap
ExecStart=/usr/local/upsnap/bin/upsnap serve --dir=/usr/local/upsnap/var/upsnap --http=0.0.0.0:8090
WorkingDirectory=/usr/local/upsnap
Restart=on-failure

[Install]
WantedBy=multi-user.target

```

**Comment by seriousm4x** [comment]
I can imagine that the feature will result in a lot of unwanted browser tabs, as pings can sometimes fail due to packet loss or other problems (e.g. Wifi). 🤔

**Comment by npelov** [comment]
Hmm. I don't understand. What I suggest is:
1. You click to power on a machine
2. Once the service is online redirect to the service url. With this feature I would prefer to open in the same tab, not new tab. So maybe also adding checkbox "same tab" would be nice
3. if power on fails (timeout) the redirect is canceled (until next wake up).

That is **only when you click** to power the machine, not if the service goes offline because of packet loss and then comes back on. 

The idea is that you power a service that you want to use - like a NAS or a web server and you see the web interface as soon as it's available. It's not that big deal to manually monitor the status until it gets green and then click on the link, but it would be nice to have.

**Reply by seriousm4x** [reply]
I see, now I understand you. I thought open a new tab once the status changes. No matter if the power button was pressed. I like the idea. I will add it to my to do list.

**Reply by seriousm4x** [reply]
Better late than never https://github.com/seriousm4x/UpSnap/commit/0e427585131763300f8b82cd013d510af8c61d0d

**Comment by npelov** [comment]
Good enough! In open source software it usually takes years to get what you want if you can't pay. I'll check this whenever I can.
